### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by stayst**
Fixed login and registration pages, added reset password page.

### CODE REVIEW

**Review by JuliaNortman — APPROVED** [review]

File: marketplace-client/src/marketplace/src/app/account/login/login.component.ts [hunk 1]
```diff
@@ -1,47 +1,13 @@
-import { Component, OnInit } from '@angular/core';
+import { Component } from '@angular/core';
 import { Router, ActivatedRoute } from '@angular/router';
 import {AbstractControl, FormBuilder, FormGroup, Validators} from '@angular/forms';
 import { first } from 'rxjs/operators';
 
-import { AccountService } from '../../_services/account.services';
+import { AccountService } from '../../_services/account.service';
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/account/login/login.component.ts:6]: unused imports

File: marketplace-client/src/marketplace/src/app/account/forgot-password/forgot-password.component.ts [hunk 2]
```diff
@@ -35,12 +39,19 @@ export class ForgotPasswordComponent{
...
+    this.accountService.resetPassword(this.form.value)
       .pipe(first())
-      .pipe(finalize(() => this.loading = false));
-    // .subscribe({
-    // next: () => this.alertService.success('Please check your email for password reset instructions'),
-    // error: error => this.alertService.error(error)
-    // });
+      .subscribe({
+        next: () => {
+          // this.router.navigate(['../registration-greeting', {relativeTo: this.route}]);
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/account/forgot-password/forgot-password.component.ts:46]: it's not ok to leave commented out code

File: marketplace-client/src/marketplace/src/app/_auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    return !this.isExpired(this.token);
+  }
+
+  public isExpectedRole(expectedRole: Role): boolean {
+    if (!this.token){
+      return false;
+    }
+    const decodedToken = decode<Token>(this.token);
+    return decodedToken.role ===  expectedRole;
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/_auth/auth.service.ts:30]: use autoformat hotkeys in your IDE

File: marketplace-client/src/marketplace/src/app/_auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  public isExpired(token: string): boolean {
+    if (this.jwtHelper.isTokenExpired(token)) { localStorage.removeItem('token'); }
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/_auth/auth.service.ts:34]: it's a bad fashion to keep it single-line

File: marketplace-client/src/marketplace/src/app/_auth/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import Token from '../_models/jwt';
+import {Role} from '../_models/role';
+import decode from 'jwt-decode';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+
+  private readonly token: string | null;
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/_auth/auth.service.ts:12]: can be just `string`

File: marketplace-client/src/marketplace/src/app/account/_components/login.form.component.html [hunk 6]
```diff
@@ -0,0 +1,45 @@
+<form [formGroup]="form" (ngSubmit)="onSubmit()">
+  <div class="form-group">
+    <label>Email</label>
+    <input type="text" formControlName="email" class="form-control"
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/account/_components/login.form.component.html:4]: it's more readable to keep 1 line = 1 attr

File: marketplace-client/src/marketplace/src/app/_services/account.service.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    return this.http.post(`${baseUrl}/register`, account);
+  }
+
+  resetPassword(email: string): Observable<any> {
+    return this.http.post(`${baseUrl}/reset-password`, email);
+  }
+
+  setNewPassword(id: string, password: string): Observable<any> {
+    const body = new ResetPasswordDTO(id, password);
+    console.log(body);
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/_services/account.service.ts:54]: is it secure to log a new password in the console?

File: marketplace-client/src/marketplace/src/app/account/reset-password/reset-password.component.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+       validator: [validateConfirmPassword, validatePassword]
+    });
+  }
+
+  get getForm(): { [p: string]: AbstractControl } { return this.form.controls; }
+
+  onSubmit(): void {
+    this.submitted = true;
+    this.loading = true;
+    console.log('here');
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/account/reset-password/reset-password.component.ts:42]: not so informative logs to be commited

File: marketplace-client/src/marketplace/src/app/account/reset-password/reset-password.component.ts [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    if (this.form.invalid) {
+      return;
+    }
+    this.route.queryParamMap.subscribe(params => {
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/account/reset-password/reset-password.component.ts:46]: do we need to unsubscribe later?

File: marketplace-client/src/marketplace/src/app/account/reset-password/reset-password.component.ts [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+      const id = params.get('id');
+      if (id) {
+        this.accountService.setToken(this.route);
+        this.accountService.setNewPassword(id, this.form.get('password')?.value)
+          .pipe(first())
+          .subscribe({
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/account/reset-password/reset-password.component.ts:52]: it's a bad fashion and can cause potential memory leaks to have nested subscriptions. It's better to use higher-order RxJs operators (like https://blog.angular-university.io/rxjs-higher-order-mapping/)

File: marketplace-client/src/marketplace/src/app/account/register/register.component.html [hunk 11]
```diff
@@ -1,53 +1,97 @@
...
-          <div class="form-group col-12">
+          <div class="form-group col-6">
             <label>First Name*</label>
             <input type="text" formControlName="name" class="form-control" [ngClass]="{ 'is-invalid': submitted && getForm.name.errors }" />
             <div *ngIf="submitted && getForm.name.errors" class="invalid-feedback">
               <div *ngIf="getForm.name.errors.required">First Name is required</div>
             </div>
           </div>
-          <div class="form-group col-12">
+          <div class="form-group col-6">
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/account/register/register.component.html:13]: use proper line indents to display tag nesting

**Review by MaxxMorozov — COMMENTED** [review]
decent FE, well done

**Pull request merged**